browser/brave_rewards/add_funds_popup.cc

```diff
@@ -0,0 +1,474 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/brave_rewards/add_funds_popup.h"
+
+#include <string>
+
+#include "base/base64.h"
+#include "base/json/json_writer.h"
+#include "base/values.h"
+#include "brave/browser/ui/views/location_bar/brave_location_bar_view.h"
+#include "brave/components/brave_rewards/browser/rewards_service.h"
+#include "brave/components/brave_shields/common/brave_shield_constants.h"
+#include "brave/components/toolbar/constants.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "chrome/browser/ui/views/frame/browser_view.h"
+#include "components/content_settings/core/browser/host_content_settings_map.h"
+#include "components/guest_view/browser/guest_view_base.h"
+#include "content/public/browser/page_navigator.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_contents_delegate.h"
+#include "content/public/common/referrer.h"
+#include "net/base/escape.h"
+#include "third_party/blink/public/platform/web_referrer_policy.h"
+#include "ui/base/page_transition_types.h"
+#include "ui/display/display.h"
+#include "ui/display/screen.h"
+#include "ui/views/widget/widget.h"
+
+using content::WebContents;
+
+namespace {
+
+  constexpr int kPopupPreferredHeight = 800;
+  constexpr int kPopupPreferredWidth = 1100;
+
+  // Referrer.
+  const char kRewardsHost[] = "rewards";
+
+  // URL to open in the popup.
+  const char kAddFundsUrl[] = "https://uphold-widget.herokuapp.com/index.php";
+
+  // Content permission URLs and patterns.
+  const char kBraveHost[] = "https://uphold-widget.herokuapp.com/";
+  const char kFirstParty[] = "https://firstParty";
```

Comment:
is this a standard URL we use?

Reply:
Yes, this is how shields distinguish all vs 3rd party: browser/ui/webui/settings/default_brave_shields_handler.cc